docs/USAGE-GUIDE.md

```diff
@@ -495,6 +501,22 @@ To use OCSP stapling, both server and client must call `s2n_config_set_status_re
 
 The OCSP stapling information will be automatically validated if the underlying libcrypto supports OCSP validation. `s2n_config_set_check_stapled_ocsp_response()` can be called with "0" to turn this off. Call `s2n_connection_get_ocsp_response()` to retrieve the received OCSP stapling information for manual verification.
 
+#### CRL Validation
+
+> Note: the CRL validation feature in s2n-tls is currently considered unstable, meaning the CRL APIs are subject to change in a future release. To access the CRL APIs, include `api/unstable/crl.h`.
+
+Certificate Revocation Lists (CRLs) are lists of issued, unexpired certificates that have been revoked by the CA. CAs publish updated versions of these lists periodically. A peer wishing to verify a certificate can obtain a CRL from the CA, validate the CRL, and check to ensure the given certificate’s serial number is not contained in the list, and thus has not been revoked by the CA.
+
+The s2n CRL lookup callback must be implemented and set via `s2n_config_set_crl_lookup_cb()` to enable CRL validation in s2n-tls. This callback will be triggered once for each certificate in the peer's certificate chain. 
+
+Applications must retrieve all CRLs for known certificates externally from s2n-tls. Applications should load the desired CRLs into memory, either in advance or in the callback, by creating `s2n_crl`s via `s2n_crl_new()`, and adding the obtained CRL data via `s2n_crl_load_pem()`. The `s2n_crl` should be freed via `s2n_crl_free()` when no longer needed.
+
+In the callback, call `s2n_crl_lookup_set()` to provide s2n-tls with a CRL to validate the certificate with. The application must implement a way to lookup the correct CRL for the given certificate. This lookup should use the certificate and CRL's issuer hashes, which will match. The certificate's issuer hash can be retrieved via `s2n_crl_lookup_get_cert_issuer_hash()`, and the CRL's issuer hash can be retrieved via `s2n_crl_get_issuer_hash()`.
+
+Call `s2n_crl_lookup_ignore()` to ignore a specific certificate in the callback. Certificate validation will fail with a `S2N_ERR_CRL_LOOKUP_FAILED` error if the ignored certificate is necessary for the chain of trust. Some received certificates may be extraneous, in that they are additional certificates that aren't ultimately included in the chain of trust used to validate the end-entity certificate. If such a certificate is ignored from the callback, certificate validation will proceed with no error.
```

Comment:
Change this paragraph to describe what to do if you can't find a CRL for the cert. Mention extraneous certs as well.

Reply:
I tried to make this more clear by describing the general case of what to do when a CRL can't be found, and then added sort of as an aside what happens when an ignored certificate is extraneous. Let me know it it's still unclear!